### ISSUE TITLE
Constraint relaxation for the epsilons

### ISSUE THREAD

**Issue by SGeeversAtVortech** [Deltares/rtc-tools#1060]
In GitLab by @TPiovesan on Aug 30, 2018, 18:20

Currently `constraint_relaxation` is used to slack the evaluation of the epsilon, so the slack that it adds is proportional to the value of the absolute difference between the bounds and the target.  So, for example, say `constrain_relaxation = 1e-6` and `function_nominal = 1`
*  if `|bound - target| = 1e6` then a slack of `1` is added to the constrain
*  if `|bound - target| = 1` then a slack of `1e-6` is added to the constrain

I would prefer if the amount of slack added by the `constraint_relaxation` is constant and not goal-dependent. The reasoning being that one can  always use `goal.relaxation` to add a considerable slack if a certain goal needs it. Moreover, in this way one has full control on how `constraint_relaxation` is acting and while setting up an optimization problem one can use information regarding the tolerance of the solver to decide what a reasonable `constraint_relaxation` should be.

In practice, this would be equivalent to do:
```
m -= options['constraint_relaxation]
M += options['constraint_relaxation]
```
and similarly for the goals. 

@baayen: thoughts?

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Dec 16, 2018, 20:24

Addressed in deae0c3f55ecf474c7d61a1d72f471d90368666f and cb59b9edccdebe38c5f9f21f9917dd5e27511b84. 
Former constraint relaxation is now "violation relaxation", and constraint relaxation now means an absolute slack.

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Dec 16, 2018, 20:24

closed
